src/pip/_internal/cli/progress_bars.py

```diff
@@ -72,7 +77,7 @@ def write_progress(current: int, total: int) -> None:
 
     write_progress(current, total)
     for chunk in iterable:
-        current += len(chunk)
+        current += unit_size or len(chunk)
```

Comment:
Should this consider the case where `unit_size` is 0?

Reply:
0 is falsy, so if someone passes in a 0 it should fall back to using the length. Unless you mean that shouldn't be the case, in which case I can change it to a None check

Reply:
So there's no difference between 0 and None? Why does this need the None case if so?

Reply:
`is None` is a different check from just a falsiness check. The code that's there will use the `len(chunk)` on any falsy value passed into unit_size. This includes False, 0, None, and a bunch of other things. So the current code will fall back to `len(chunk)` if someone were to pass 0 into this (which is what I assumed you wanted me to write an explicit case for)

If that _isn't_ what you were requesting, and you _do_ want 0 to be a valid value for unit_size (I'm not really sure what the use case of that would be if so), I would have to replace that with something like `len(chunk) if unit_size is None else unit_size`. Alternatively I could put an assertion or an if statement explicitly checking to make sure the value is > 0 and throw an error if it isn't, rather than just using the chunk length.

Reply:
So is there something that still needs action for this?

Reply:
I would prefer this to default to 0 and drop the None case altogether (both in annotation and handling). But this is an internal function so I won’t insist if you prefer the current style.

Reply:
So instead of using the length of the provided value if the override length isn't provided, you'd like me to use 0? The progress would never increase in that case

Reply:
No… make `unit_size` default to 0 (instead of None), and coerce it to `len(chunk)` in the function.

Reply:
Ahhhhhh ok, I understand now